state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -757,7 +759,15 @@ pub mod pallet {
 				id,
 				option_tick,
 				sell_amount,
-			)
+			)?;
+
+			if let Some(expire_at) = expire_at {
+				let call = Call::cancel_limit_order { base_asset, quote_asset, side, id };
+				Self::schedule_limit_order_update(origin, Box::new(call), expire_at)
```

Comment:
This function checks if the expiry_at block is greater or equal than the current block. But for this particular case, it doesn't make much sense: If expire_at is equal to the current block, we would create the order and then immediately remove it. 

We should add a check to ensure that expire_at is greater than the current block.

Reply:
Btw. I might make sense to break `schedule_limit_order_update` out into a pallet method instead of calling the extrinsic directly, otherwise we're duplicating things like the origin check.